cores/esp8266/Arduino.h

```diff
@@ -233,25 +240,22 @@ const int TIM_DIV265 __attribute__((deprecated, weak)) = TIM_DIV256;
 
 
 
+// from this point onward, we need to configure the c++ environment
 #ifdef __cplusplus
 
 #include <algorithm>
+#include <cstdlib>
 #include <cmath>
-#include <pgmspace.h>
-
-#include "WCharacter.h"
-#include "WString.h"
-
-#include "HardwareSerial.h"
-#include "Esp.h"
-#include "Updater.h"
-#include "debug.h"
 
 using std::min;
 using std::max;
 using std::isinf;
 using std::isnan;
 
+// these are important, as we may end up using C versions otherwise
+using std::abs;
+using std::round;
```

Comment:
Meaning, right now anyone using both of these uses C stdlib versions, undef + define never did anything because of the cstdlib include in algorithm